client/src/backoffice/EventsPage.tsx

```diff
@@ -169,44 +174,101 @@ const EventField: React.FC<EventFieldProps> = ({ title, name, disabled }) => {
 
 const EventList: React.FC = () => {
   const [events, fetchingEvents, fetchEventsError] = useAsync(getEvents);
+  const [criteria, setCriteria] = useState<string>('');
 
   return (
-    <div className={"bk-container"}>
+    <div className={'bk-container'}>
       <h2>Events</h2>
       <Link to="/admin/events/new">
         <button className="bk-btn" style={{ margin: '30px 0px' }}>
           Create New
         </button>
       </Link>
-      {fetchingEvents ? (
-        <div>Fetching Events...</div>
-      ) : fetchEventsError || events == null ? (
-        <div>There was a problem fetching events</div>
-      ) : (
-        <div>
-          {events.map(e => (
-            <EventRow key={e.id} event={e} />
-          ))}
-        </div>
-      )}
+      <input
+        type="text"
+        placeholder="Search by name"
+        onChange={e => setCriteria(e.target.value.toLowerCase())}
```

Comment:
Prefer avoiding inline functions and add a const outside of the render and short parameter naming.

Code example:
```typescript
const handleNameChange = (event: React.FormEvent<HTMLInputElement>): void => setCriteria(event.target.value.toLowerCase());
```